javalib/src/main/scala/java/util/AbstractSequentialList.scala

```diff
@@ -0,0 +1,42 @@
+package java.util
+
+abstract class AbstractSequentialList[E] protected ()
+    extends AbstractList[E] {
+
+  def get(index: Int): E =
+    listIterator(index).next
+
+  override def set(index: Int, element: E): E = {
+    val iter = listIterator(index)
+    val ret = iter.next
+    iter.set(element)
+    ret
+  }
+
+  override def add(index: Int, element: E): Unit =
+    listIterator(index).add(element)
+
+  override def remove(index: Int): E = {
+    val iter = listIterator(index)
+    val ret = iter.next
+    iter.remove
+    ret
+  }
+
+  override def addAll(index: Int, c: Collection[_ <: E]): Boolean = {
+    val iter = listIterator(index)
+    val citer = c.iterator
+    var changed = false
+    while (citer.hasNext) {
+      iter.add(citer.next)
+      iter.next
+      changed = true
+    }
+    changed
```

Comment:
I actually prefer the `while` version. When working with an iterator, I find it disturbing to use `foldLeft`. In general, I find it disturbing to use anything side-effecting in the function given to `foldLeft` (or `forall` or things like that). We are dealing with side-effects anyway. We might as well make that very explicit with a `while` loop.

Note that in this case, It's actually a good old `foreach` on `citer` I believe.